support-lambdas/stripe-intent/src/main/scala/com/gu/handler/impure/S3Loader.scala

```diff
@@ -0,0 +1,44 @@
+package com.gu.handler.impure
+
+import com.amazonaws.auth.profile.ProfileCredentialsProvider
+import com.amazonaws.auth.{AWSCredentialsProviderChain, EnvironmentVariableCredentialsProvider, InstanceProfileCredentialsProvider}
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.s3.model.S3Object
+import com.amazonaws.services.s3.{AmazonS3ClientBuilder, AmazonS3URI}
+import com.gu.monitoring.SafeLogger
+import com.typesafe.config.{Config, ConfigFactory}
+
+import scala.io.{BufferedSource, Source}
+
+object Aws {
+
+  val ProfileName = "membership"
+
+  lazy val CredentialsProvider = new AWSCredentialsProviderChain(
+    new ProfileCredentialsProvider(ProfileName),
+    new InstanceProfileCredentialsProvider(false),
+    new EnvironmentVariableCredentialsProvider()
+  )
+
+}
+
+object S3Loader {
+  def load(uri: AmazonS3URI): Config = {
+    SafeLogger.info(s"Loading config from S3 from $uri")
+    val s3Client = AmazonS3ClientBuilder
+      .standard()
+      .withCredentials(Aws.CredentialsProvider)
+      .withRegion(Regions.EU_WEST_1)
+      .build()
+
+    val s3Object: S3Object = s3Client.getObject(uri.getBucket, uri.getKey)
+
+    val source: BufferedSource = Source.fromInputStream(s3Object.getObjectContent)
+    try {
+      val conf = source.mkString
+      ConfigFactory.parseString(conf).withFallback(ConfigFactory.load())
+    } finally {
+      source.close()
```

Comment:
you should also call `s3Object.close` here

Reply:
you mean just for completeness? as looking at the source code closing the Source would close the stream, and calling close on the s3 object does the same.

Reply:
oh good point, I thought it did more than that!

Reply:
even the docs are not so clear. https://docs.aws.amazon.com/AmazonS3/latest/dev/RetrievingObjectUsingJava.html

Reply:
indeed, but the actual code is clear!